Add a "Check that everything is fine" page

fxthomas opened the pull request:
This is a first attempt with some of the ideas in #1082, implemented just like the "About" page.

It's not finished yet, but it should be functional enough for testing so I'll leave it here for comments.

Screenshot:

![Screen Shot 2020-01-05 at 18 06 08-fullpage](https://user-images.githubusercontent.com/613594/71783387-67e0f100-2fe6-11ea-991e-8206f26157a6.jpg)

jvoisin commented on airsonic-main/src/main/java/org/airsonic/player/controller/InternalHelpController.java:
```diff
@@ -0,0 +1,299 @@
+            map.put("indexSongCount", reader.numDocs());
+            map.put("indexSongDeletedCount", reader.numDeletedDocs());
+        } catch (IOException e) {
+            LOG.debug("Unable to gather information", e);
```
This should be changed for less generic log message :)

jvoisin commented:
This looks amazing!

fxthomas commented:
Thanks :wink: I added info about ffmpeg + ffprobe in addition to a small refactoring.

![Screenshot from 2020-01-07 22-49-45](https://user-images.githubusercontent.com/613594/71932638-8559ba80-31a0-11ea-840b-abd597bab90c.png)

Now this just needs tweaking log messages, adding the correct i18n and testing on other databases (not sure if what I did works on PostgreSQL/MySQL).

fxthomas commented:
I moved a few things around, tested on PostgreSQL, added i18n and a few additional consistency checks.

Should be good for review and testing now!

jvoisin commented:
This looks glorious, and it would be a shame to let it rot: can we merge it?

fxthomas commented:
Thanks!

I did some more tests on this to make sure it's going to be robust on recent PostgreSQL/MariaDB/MySQL versions, hope you don't mind the last minute bits:

* 27a3a7e: Don't check db size on filesystem in non-`legacy` profiles (we don't know how to do that in a consistent manner)
* 8734aa6: Fix a crash with MariaDB (which I hadn't tested earlier)
* 3e12c71: Add the server version in the page
* 6c8d133: Add simple tests for admin users and non-admin users

All previous commits are the same as what you reviewed, squashed and rebased on master.

fxthomas commented:
With that it should be fully tested on all 3 major databases and ready to merge whenever you're okay with it.

fxthomas commented:
Added one more last minute fix for the last failure. Obviously I should know that saying "everything is good" before Travis runs is a huge death flag... :rofl: 

Not sure why it's complaining about `cxf-gen` though?

fxthomas commented:
I cleared Travis's cache, hopefully this fixes the build?
